Example/wifi_sta/firmware/src/app.c

```diff
@@ -75,7 +114,176 @@ uint8_t CACHE_ALIGN work[SYS_FS_FAT_MAX_SS];
 
 
 /* TODO:  Add any necessary local functions.
-*/
+ */
+
+
+bool validateUserPass (char * user, char * pass){
+    
+    if ((strlen(user) > 10) || (strlen(pass) > 10)) {
+        
+        SYS_CONSOLE_MESSAGE("Username or Password is too long\n\r");
+        return false;
+    }
+
+    return true;
+}
+
+bool FTP_Command_Initialize(void){
+    
+    if (!SYS_CMD_ADDGRP(ftpCmdTbl, sizeof(ftpCmdTbl)/sizeof(*ftpCmdTbl), "ftp", ": commands"))
+    {
+        SYS_ERROR(SYS_ERROR_ERROR, "Failed to create FTP Commands\r\n");
+        return false;
+    }
+}
+
+static inline void _Command_setECC(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    setECC = true;                          //ECC will be used to store FTP server login credentials
+    appData.state = APP_ECC_Init;
+}
+
+static int _Command_AddUser(SYS_CMD_DEVICE_NODE* pCmdIO, int argc, char** argv){
+    
+    if(setECC){
+        clientLoginInfo client;
+        
+        strncpy(client.username, argv[1], 10);
+        strncpy(client.password, argv[2], 10);
+        if(validateUserPass(client.username, client.password))
```

Comment:
If the username length is 10 bytes, then validateUserPass strlen(user) will be added with the password length.
The username and password buffer are contagious memory and strncpy copies 10 bytes without the space for the null string character.

Reply:
removed the function